tensorboard/plugins/scalar/scalars_plugin.py

```diff
@@ -129,7 +150,19 @@ def index_impl(self):
 
   def scalars_impl(self, tag, run, experiment, output_format):
     """Result of the form `(body, mime_type)`."""
-    if self._db_connection_provider:
+    if self._data_provider:
+      all_scalars = self._data_provider.read_scalars(
+          experiment_id=None,  # experiment support not yet implemented
+          owner_plugin=metadata.PLUGIN_NAME,
+          downsample_to=1000,
+          run_tag_filter=provider.RunTagFilter(runs=[run], tags=[tag]),
+          step_filter=None,  # also the default
+      )
+      scalars = all_scalars.get(run, {}).get(tag, None)
+      if scalars is None:
+        raise KeyError('No scalar data for run=%r, tag=%r' % (run, tag))
```

Comment:
KeyError seems like a bit of a weird exception to continue using as a holdover.  Can we maybe user a better one in the new code and just just pass a tuple of that and KeyError to assertRaises() in the tests?

Reply:
Using `KeyError` seems fine with me (“Mapping key not found” is its
docstring), but I suppose that `ValueError` could also be appropriate.

No strong opinion given that the HTTP endpoint _should_ project it down
to a 400 Bad Request anyway (though it currently just 500s).

Reply:
Changed to `ValueError`.